.circleci/workflows.yml

```diff
@@ -0,0 +1,1151 @@
+---
+version: 2.1
+
+orbs:
+  gcp-gcr: circleci/gcp-gcr@0.13.0
+  docker: circleci/docker@2.2
+  python: circleci/python@2.1.1
+
+parameters:
+  python-version:
+    type: string
+    default: '3.10'
+  validate-bqetl:
+    type: boolean
+    default: false
+  validate-sql:
+    type: boolean
+    default: false
+  validate-routines:
+    type: boolean
+    default: false
+  deploy:
+    type: boolean
+    default: false
+  trigger-sql-generation:
+    type: boolean
+    default: false
+  skip-stage-deploys:
+    type: boolean
+    default: false
+
+executors:
+  ubuntu-machine-executor:
+    machine:
+      image: ubuntu-2004:202111-02
+
+jobs:
+  build:
+    docker: &docker
+      - image: python:<< pipeline.parameters.python-version >>
+    steps:
+      - checkout
+      - &restore_venv_cache
+        restore_cache:
+          keys:
+            # when lock files change, use increasingly general
+            # patterns to restore cache
+            - &python_cache_key
+              # yamllint disable-line rule:line-length
+              python-<< pipeline.parameters.python-version >>-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-{{ checksum "requirements.txt" }}
+            # yamllint disable-line rule:line-length
+            - python-<< pipeline.parameters.python-version >>-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-
+            # yamllint disable-line rule:line-length
+            - python-<< pipeline.parameters.python-version >>-packages-v1-{{ .Branch }}-
+            - python-<< pipeline.parameters.python-version >>-packages-v1-main-
+      - &build
+        run:
+          name: Build
+          command: |
+            python3 -m venv venv/
+            venv/bin/pip install --no-deps -r requirements.txt
+            venv/bin/pip-sync --pip-args=--no-deps
+      - save_cache:
+          paths:
+            - venv/
+          key: *python_cache_key
+  verify-format-sql:
+    docker: *docker
+    steps:
+      - when:
+          condition:
+            or:
+              - << pipeline.parameters.validate-sql >>
+              - << pipeline.parameters.validate-routines >>
+          steps:
+            - checkout
+            - *restore_venv_cache
+            - *build
+            - run:
+                name: Verify that SQL is correctly formatted
+                command: |
+                  PATH="venv/bin:$PATH" script/bqetl format --check \
+                  $(git ls-tree -d HEAD --name-only)
+      - run: exit 0
```

Comment:
CircleCI needs to have at least one `run` step, so adding these as dummy steps

Reply:
Maybe instead of having an always-run no-op step this could instead be an `unless` conditional, using the same condition as the `when` (e.g. via YAML reference to keep it DRY), with a single step named something meaningful like "skipped" (and again could use YAML references to avoid repeating the same skipped/no-op step code everywhere)?